diff --git a/device/mediatek/sepolicy/basic/neverallows/non_plat/neverallows.te b/device/mediatek/sepolicy/basic/neverallows/non_plat/neverallows.te
index 7e7f7ab5c73..5a80453d613 100755
--- a/device/mediatek/sepolicy/basic/neverallows/non_plat/neverallows.te
+++ b/device/mediatek/sepolicy/basic/neverallows/non_plat/neverallows.te
@@ -36,7 +36,6 @@ full_treble_only(`
     ueventd
     vendor_init
     vold
-    btlfpserver
     factory
     hal_paxservice_default
     system_server
@@ -97,7 +96,6 @@ full_treble_only(`
     system_server
     vendor_init
     vold
-    btlfpserver
     } proc:file *;
 
   neverallow {
diff --git a/device/mediateksample/k62v1_64_pax/ProjectConfig.mk b/device/mediateksample/k62v1_64_pax/ProjectConfig.mk
index ef1048a5d7a..70f548c34e8 100755
--- a/device/mediateksample/k62v1_64_pax/ProjectConfig.mk
+++ b/device/mediateksample/k62v1_64_pax/ProjectConfig.mk
@@ -381,7 +381,7 @@ MTK_TB_APP_LANDSCAPE_SUPPORT = no
 MTK_TC1_FEATURE = no
 MTK_TDD_DATA_ONLY_SUPPORT = no
 MTK_TEE_GP_SUPPORT = no
-MTK_TEE_SUPPORT = yes
+MTK_TEE_SUPPORT = no
 MTK_TELEPHONY_CONN_LOG_CTRL_SUPPORT = yes
 MTK_TELEPHONY_FEATURE_SWITCH_DYNAMICALLY = no
 MTK_TELEPHONY_MODE = 0
@@ -467,8 +467,8 @@ MTK_CAM_BGSERVICE_SUPPORT = yes
 MTK_VOW_BARGE_IN_SUPPORT = no
 MTK_NETWORK_MONITOR = no
 MTK_WIFI_DUAL_BAND_SUPPORT = yes
-TRUSTKERNEL_TEE_SUPPORT = yes
-TRUSTKERNEL_TEE_PLATFORM_APP_SUPPORT = yes
+TRUSTKERNEL_TEE_SUPPORT = no
+TRUSTKERNEL_TEE_PLATFORM_APP_SUPPORT = no
 MTK_TEE_DRAM_SIZE = 0x2400000
 MTK_HEIF_CAPTURE_SUPPORT = no
 MTK_BT_SAP_ENABLE = yes
@@ -553,8 +553,8 @@ MTK_CAM_TABLET_CCU_SUPPORT = yes
 MTK_CAM_SINGLE_NR_SUPPORT = yes
 MTK_TOUCHPANEL_KO =
 MTK_FAST_CHARGER_TECH = no
-MTK_FINGERPRINT_SUPPORT = yes
-MTK_SUPPORT_MULTI_FINGERPRINT_VENDOR = yes
+MTK_FINGERPRINT_SUPPORT = no
+MTK_SUPPORT_MULTI_FINGERPRINT_VENDOR = no
 ##########HXY ADD BEGIN###########
 
 #功能:GMS core app 核心app的开关
diff --git a/kernel-4.19/arch/arm64/boot/dts/mediatek/M8.dts b/kernel-4.19/arch/arm64/boot/dts/mediatek/M8.dts
index 35f1e3ffb75..243f57803b5 100755
--- a/kernel-4.19/arch/arm64/boot/dts/mediatek/M8.dts
+++ b/kernel-4.19/arch/arm64/boot/dts/mediatek/M8.dts
@@ -690,7 +690,7 @@
 };
 
 &pax_gpios {
-	status = "okay";
+	status = "disabled";
 	gpio_num = <9>;
 	gpio_in_num = <4>;
 	gpio_input_index_array = <0 2 3 4>;
diff --git a/kernel-4.19/arch/arm64/configs/k62v1_64_pax_debug_defconfig b/kernel-4.19/arch/arm64/configs/k62v1_64_pax_debug_defconfig
index 43838a8a23d..1d9f88d8b20 100755
--- a/kernel-4.19/arch/arm64/configs/k62v1_64_pax_debug_defconfig
+++ b/kernel-4.19/arch/arm64/configs/k62v1_64_pax_debug_defconfig
@@ -518,16 +518,16 @@ CONFIG_MTK_FTRACE_DEFAULT_ENABLE=y
 # CONFIG_RUNTIME_TESTING_MENU is not set
 CONFIG_CORESIGHT=y
 CONFIG_CORESIGHT_STM=y
-CONFIG_MTK_FINGERPRINT_SUPPORT=y
+# CONFIG_MTK_FINGERPRINT_SUPPORT is not set
 CONFIG_MTK_FINGERPRINT_CHIPSAILING_CS1501=y
 CONFIG_BETTERLIFE_FINGERPRINT=y
 # pax
 CONFIG_PAX_DRV_SUPPORT=y
 CONFIG_PAX_GPIOS_CONTROL=y
 #TEE
-CONFIG_TEE=y
-CONFIG_TRUSTKERNEL_TEE_SUPPORT=y
-CONFIG_TRUSTKERNEL_TEE_FP_SUPPORT=y
+# CONFIG_TEE is not set
+# CONFIG_TRUSTKERNEL_TEE_SUPPORT is not set
+# CONFIG_TRUSTKERNEL_TEE_FP_SUPPORT is not set
 CONFIG_PAX_CAM_REGULATOR_SUPPORT=y
 CONFIG_SND_SOC_AW87XXX=y
 CONFIG_PAX_USB_SWITCH=y
diff --git a/kernel-4.19/arch/arm64/configs/k62v1_64_pax_defconfig b/kernel-4.19/arch/arm64/configs/k62v1_64_pax_defconfig
index 9bbfb39ab2d..dcbfeb9122b 100755
--- a/kernel-4.19/arch/arm64/configs/k62v1_64_pax_defconfig
+++ b/kernel-4.19/arch/arm64/configs/k62v1_64_pax_defconfig
@@ -496,16 +496,16 @@ CONFIG_MTK_FTRACER=y
 # CONFIG_RUNTIME_TESTING_MENU is not set
 CONFIG_CORESIGHT=y
 CONFIG_CORESIGHT_STM=y
-CONFIG_MTK_FINGERPRINT_SUPPORT=y
+# CONFIG_MTK_FINGERPRINT_SUPPORT is not set
 CONFIG_MTK_FINGERPRINT_CHIPSAILING_CS1501=y
 CONFIG_BETTERLIFE_FINGERPRINT=y
 # pax
 CONFIG_PAX_DRV_SUPPORT=y
 CONFIG_PAX_GPIOS_CONTROL=y
 #TEE
-CONFIG_TEE=y
-CONFIG_TRUSTKERNEL_TEE_SUPPORT=y
-CONFIG_TRUSTKERNEL_TEE_FP_SUPPORT=y
+# CONFIG_TEE is not set
+# CONFIG_TRUSTKERNEL_TEE_SUPPORT is not set
+# CONFIG_TRUSTKERNEL_TEE_FP_SUPPORT is not set
 CONFIG_PAX_CAM_REGULATOR_SUPPORT=y
 CONFIG_SND_SOC_AW87XXX=y
 CONFIG_PAX_USB_SWITCH=y
diff --git a/kernel-4.19/drivers/tty/serial/8250/8250_core.c b/kernel-4.19/drivers/tty/serial/8250/8250_core.c
index fb089f57dcd..185497772ec 100755
--- a/kernel-4.19/drivers/tty/serial/8250/8250_core.c
+++ b/kernel-4.19/drivers/tty/serial/8250/8250_core.c
@@ -732,7 +732,7 @@ static int __init univ8250_console_init(void)
 	//	if (!cmdline_get_terminal_name(terminal_name)) {
 	//		if (!strncmp(terminal_name, "M50", 3)) {
 				//#ifndef CONFIG_MTK_ENG_BUILD
-				univ8250_console.flags = CON_BOOT;
+	//			univ8250_console.flags = CON_BOOT;
 				//#endif
 	//		}
 	//	}
diff --git a/vendor/mediatek/proprietary/bootable/bootloader/preloader/custom/k62v1_64_pax/k62v1_64_pax.mk b/vendor/mediatek/proprietary/bootable/bootloader/preloader/custom/k62v1_64_pax/k62v1_64_pax.mk
index 9603e46a59a..bb46edf9d85 100755
--- a/vendor/mediatek/proprietary/bootable/bootloader/preloader/custom/k62v1_64_pax/k62v1_64_pax.mk
+++ b/vendor/mediatek/proprietary/bootable/bootloader/preloader/custom/k62v1_64_pax/k62v1_64_pax.mk
@@ -26,8 +26,8 @@ MTK_TINYSYS_SSPM_SUPPORT=yes
 MTK_UFS_OTP_SUPPORT=no
 MTK_UFS_POWP_SUPPORT=no
 RPMB_PER_DEVICE_KEY=no
-MTK_TEE_SUPPORT = yes
-TRUSTKERNEL_TEE_SUPPORT = yes
+MTK_TEE_SUPPORT = no
+TRUSTKERNEL_TEE_SUPPORT = no
 TRUSTONIC_TEE_SUPPORT = no
 MICROTRUST_TEE_SUPPORT = no
 MTK_GOOGLE_TRUSTY_SUPPORT = no
